### ISSUE TITLE
On windows and linux: Capital and lowercase 'w' have size and kerning issues ($500 bounty)

### ISSUE THREAD

**Issue by Discordius** [edwardtufte/et-bembo#20]
On windows and linux machines you get the following effect if you display the font at anything but the default size of `20px`: 

<img width=360 src=https://user-images.githubusercontent.com/154613/86421031-7878d800-bca6-11ea-857b-5c8b6020efd0.png>

This makes most of the text sadly almost unreadable on windows and linux devices.

**Comment by rtm** [comment]
Thanks for your work on this font. I'm having exactly the same problem. I don't see it at 18px (1.2rem), but it's quite noticeable at 1.1rem (16px). See screenshot.

![image](https://user-images.githubusercontent.com/24391/45772404-36171000-bc65-11e8-894c-4894f6af0b7e.png)

This pretty much excludes my using this font in my app. Should I go ahead and find something else, or is there any chance this could be addressed?

**Comment by adamschwartz** [comment]
This seems like roughly the same issue as https://github.com/edwardtufte/et-book/issues/19. Closing as duplicate for now.

**Comment by rtm** [comment]
@adamschwartz The root cause may well be the same--or not. This bug relates to the lower-case W, and in particular its relative size, while #19 relates to kerning issues with the upper-case W.

**Comment by skosch** [comment]
I can't reproduce this on Linux. Perhaps it's a hinting issue on Windows – try running the files through ttfautohint?

**Comment by Discordius** [comment]
I reproduced it on Windows Edge 81 on this URL http://shlegeris.com/2018/10/23/weirdest, with the font-size set to 17px. 

Here is a screenshot: 

![image](https://user-images.githubusercontent.com/9090789/86501264-37d39a00-bd4c-11ea-87df-dc6273caf7e3.png)

**Comment by Discordius** [comment]
Just in case anyone stumbles across this, this is still a problem for me, and I am willing to send $500 via wire transfer or PayPal to whoever fixes this. And $100 to anyone who provides a pointer that helps someone else eventually fix it (or helps me fix it), up to a maximum of another $500 (i.e. up to 5 pointers).

**Comment by skosch** [comment]
What browser are you using in your screenshot from July 3? I'm going through combinations of Windows/Edge versions on Browserstack to try and reproduce the problem, but I can't find one that looks like yours above :thinking:

**Comment by skosch** [comment]
For instance, here's Edge 85 on Windows 10:
![image](https://user-images.githubusercontent.com/554383/93002214-3e1f8880-f4ea-11ea-9b7c-cc3d32d645f7.png)

**Comment by skosch** [comment]
Ah sorry, please disregard – I didn't see your comment about 17px. I see it now

**Comment by skosch** [comment]
Ok @Discordius, as I suspected it's a hinting issue. What you need to do is take the TTFs from the [source](https://github.com/edwardtufte/et-book/tree/gh-pages/source/4-ttf) folder and run them through a webfont generator like [this one](https://www.fontsquirrel.com/tools/webfont-generator). Select "Expert", and then you can choose from hinting presets.

For me, the "Font Squirrel" setting results in the wonky w on Windows, but "Keep Existing" and "TTFAutohint" doesn't. See https://www.aldusleaf.org/tufte/test.html below. As far as I can tell this is robust to font size changes, but you should definitely play around with it and make sure there's no other issues.

![image](https://user-images.githubusercontent.com/554383/93002854-bf791a00-f4ee-11ea-9230-ba4d2e2dd34c.png)

**Comment by Discordius** [comment]
Great! If you make a PR to this repo with the fixed `.ttf` files and it gets merged, then I am happy to send you $500 via PayPal (or if it doesn't get merged after two weeks or so, which can of course easily happen if the maintainer doesn't have time to double-check).

Otherwise I would do that myself and send you $100 for giving me a successful pointer to a potential solution. The $500 seems fairer, but the commitment I made was to only pay out the $500 for the full end-to-end thing.

**Comment by Discordius** [comment]
I will double check the for any errors that result from this. But thank you for your help in any case, and no matter what problems I find, I am happy to send you at least $100 in any case.

**Comment by skosch** [comment]
Yes, please do double check – and I think we should get @adamschwartz's input before creating new webfonts, to make sure they match the existing files as closely as possible, especially w.r.t. scaling. Fontsquirrel has a lot of settings!

Your bounty offer is really generous. I would like to see the money go to an effective charity, perhaps [ITIF](https://lets-fund.org/Clean-Energy/) in light of the current fires?

**Comment by adamschwartz** [comment]
Thanks @skosch and @Discordius 

I would be happy to see the results. I don’t have enough experience with this type of issue to say I know what the ideal solution is here, and unfortunately I don’t have a Windows machine to test this on. 

My suggestion is that once we have both fonts to compare (current and proposed), we build a test page which renders the same text in both fonts placed on top of each other (z-dimension) in various font sizes (including the troublesome 17px). Make the top one black and the bottom one red, e.g. We can easily see if any red pokes out. That would be a good start.

**Comment by Discordius** [comment]
> Your bounty offer is really generous. I would like to see the money go to an effective charity, perhaps ITIF in light of the current fires?

Of course, that seems great! Will send it out as soon as we settled stuff here, which will determine the amount. 

I like the idea of a test page with both of them rendered on top of each other.

**Comment by skosch** [comment]
Alright, as far as I can tell, there's no significant difference between the settings, as long as we avoid the "Font Squirrel" preset; see https://www.aldusleaf.org/tufte/test.html. The metrics seem to match, and that's without me touching any of the other options. Can you try and play with it some more and see if you find any issues?

Finally, this would be a good time to set up proper OpenType lookups for both ligatures and oldstyle/lining variants. This will take me a bit of time; if you'd like to go ahead and do it yourself that's fine too.

**Comment by adamschwartz** [comment]
Thanks so much. This is a great start!

The main thing that caught my eye was a subtle change to the opening inside “e”:

<img width=100 src=https://user-images.githubusercontent.com/154613/93028128-da778700-f5df-11ea-84eb-2e1c82397da6.png>

I believe I actually prefer the new one, although I wasn’t 100% sure which was which in the test. Please add a key to make sure it’s clear, and please use red beneath the black as shown in this screenshot.

There are a couple of other similar changes to the curvature of “s”, e.g.

I’d love to see the test additionally render all defined glyphs in a large font-size to make these sorts of comparisons easier. (You can use https://fontdrop.info or similar to get a good table/list to start.)

Check out https://fontdrop.info/#/compare as well.

**Comment by skosch** [comment]
I've reverted the colours on the test page, red is now the original (underneath) and black the new (on top).

Here are the five files:
* [Original](https://www.aldusleaf.org/tufte/et-book-roman-line-figures.woff)
* [Keep existing, "fix GASP table"](https://www.aldusleaf.org/tufte/et-book-existing-fixgasp.woff)
* [Keep existing, no "fix GASP table"](https://www.aldusleaf.org/tufte/et-book-existing-nofixgasp.woff)
* [TTFAutohint, "fix GASP table"](https://www.aldusleaf.org/tufte/et-book-ttfautohint-fixgasp.woff)
* [TTFAutohint, no "fix GASP table"](https://www.aldusleaf.org/tufte/et-book-ttfautohint-nofixgasp.woff)

I don't see any differences on Fontdrop, at least not on my Linux machine. Do you? And I don't know how I would use Fontdrop via Browserstack, as it's not letting me download the files within the VM ...

**Comment by adamschwartz** [comment]
I tested in Chrome and Firefox on macOS and things are looking good.

To feel confident this really works, I’d love to get confirmation from one or more Windows folks, with ideally some screenshots posted here. As far as I’m concerned, the differences in the test on macOS are negligible enough that if we’re solving a real problem with the `w` on Windows, it’s worth it to me.

@Discordius since you’re the main driver here, would you be willing to produce screenshots of the test pages on Windows?

**Comment by Discordius** [comment]
Sure! Taken from browserstack.

100% Zoom: 

![image](https://user-images.githubusercontent.com/9090789/93543730-32c8c480-f911-11ea-80bc-e5303a29ce2a.png)

125% Zoom: 

![image](https://user-images.githubusercontent.com/9090789/93543769-4f64fc80-f911-11ea-9bfe-55edb6948ec8.png)

150% Zoom: 

![image](https://user-images.githubusercontent.com/9090789/93543826-699eda80-f911-11ea-8ee2-b7baea316db9.png)
![image](https://user-images.githubusercontent.com/9090789/93543814-61df3600-f911-11ea-8227-21b3b59f9eb5.png)

200% Zoom: 

![image](https://user-images.githubusercontent.com/9090789/93543846-76233300-f911-11ea-80b2-ac94ffdc5842.png)
![image](https://user-images.githubusercontent.com/9090789/93543867-7f140480-f911-11ea-99f2-085a21505b89.png)
![image](https://user-images.githubusercontent.com/9090789/93543879-876c3f80-f911-11ea-83e1-c3482c47d72f.png)
![image](https://user-images.githubusercontent.com/9090789/93543901-905d1100-f911-11ea-9ace-34d680f53b84.png)

**Comment by adamschwartz** [comment]
Thanks so much for these. I’m curious how you’re doing the the zooming.

The lowercase “g” seems to vary quite wildly between the size. Compare the word “gone” in the first sentence (“Over the last year or two...”) in 150% and 200% for example:

150%:
![image](https://user-images.githubusercontent.com/154613/93563922-d7212a00-f956-11ea-92da-f2f97c861966.png)

200%:
![image](https://user-images.githubusercontent.com/154613/93563937-dd170b00-f956-11ea-9150-9398868c9a49.png)

It also seems like the screenshots may not have been captured and/or uploaded in native resolution. Would it be possible to redo these that way?

As these stand I am seeing a pretty significant amount of red. The lowercased “e” and “g” both seem to have some issues, given the blurriness of the screenshots, it’s hard for me to say if those issues were already present in the original version.

**Comment by skosch** [comment]
I'm not sure what to do about the *g*, I agree it doesn't look great. TTFAutohint has [lots of settings](https://www.freetype.org/ttfautohint/doc/ttfautohint.html), and [hand-hinting](https://fontforge.org/docs/techref/hinting.html) is always an option. I've never done either though and don't know how much time and experimentation it would take to get it right.

**Comment by Discordius** [comment]
> Thanks so much for these. I’m curious how you’re doing the the zooming.

I've just been zooming in with the browser

> It also seems like the screenshots may not have been captured and/or uploaded in native resolution. Would it be possible to redo these that way?

Alas, I think this is the result of me testing on Browserstack, since I also don't have a Windows machine available. With browserstack compression plus browserstack presumably not sending the feed of a retina monitor back to me.

I have colleague with a windows machine who can probably take better screenshots. I will ask him.

**Comment by adamschwartz** [comment]
Ah that’s right I forgot about Browserstack. Thanks

**Comment by sirinath** [comment]
Is it possible to add the full Latin, Greek, diacritic, and symbols, if there is a possibility?

**Comment by Discordius** [comment]
@sirinath: My guess is you should create a new issue for that, since this issue is about something much narrower.

**Comment by Discordius** [comment]
@adamschwartz: About the top-level issue that we were discussing. Is this currently blocked on having a good set of high-resolution windows screenshots? If so, then I am happy to get those.

**Comment by adamschwartz** [comment]
@sirinath Please create a separate issue for that as @Discordius noted. Thanks.

@Discordius yes, thank you!

**Comment by Discordius** [comment]
Here they are! 
![120230815_753414591882764_1687894932638394619_n](https://user-images.githubusercontent.com/9090789/94878182-86f39e80-0411-11eb-8d8c-47f0179e1bbb.png)
![120298868_254319809264644_25624287256365123_n](https://user-images.githubusercontent.com/9090789/94878183-8824cb80-0411-11eb-87f4-b2037c6458f8.png)
![120273684_1878112068996609_2058953158131584105_n](https://user-images.githubusercontent.com/9090789/94878186-8955f880-0411-11eb-93a8-f7eeca6b5822.png)
![120451994_4423521724389213_2226609821907694302_n](https://user-images.githubusercontent.com/9090789/94878188-89ee8f00-0411-11eb-9c78-ff9e19b5fea3.png)
![120569537_1043818252740893_2679360240916076442_n](https://user-images.githubusercontent.com/9090789/94878189-8bb85280-0411-11eb-8a80-11561c0a298e.png)

**Comment by adamschwartz** [comment]
Thank you so much for these.

As expected, the differences are somewhat clearer to see in the native screenshots.

My concerns around the “g” are unfortunately still there.

As for the “e”, I’m now able to see that similar issues are occurring in many other glyphs, e.g. “d”, “f”, “i”, “l”, “p”, and “u”.

![image](https://user-images.githubusercontent.com/154613/94880572-56236180-0431-11eb-8f51-ff1636c38848.png)

![image](https://user-images.githubusercontent.com/154613/94880561-4dcb2680-0431-11eb-8cb8-468d31f19d23.png)

That is to say, it seems like in general, there is a vertical stretch that seems to be occurring, exposing a horizontal bar of red in many glyphs.

That being said, these issues I think we could live with, certainly in exchange for fixing the “w”.

The lowercased “g” though is still really bothering me. I’d like to leave this open for a little bit longer, and I’d encourage us all to reach out to typographers we know to see if it can be relatively easily addressed. If within a week or two we cannot find a solution, I’m willing to merge really any of the options above to fix the “w” issue—which is more severely limiting usage on Windows and Linux. However, at that point, we’ll be closing this one and opening up a new issue to fix the “g” haha. (Still worth it.)

@skosch Do you have any other recommendations before we give up?

**Comment by skosch** [comment]
Thanks for these screenshots, I agree they're very helpful.

I'm not as worried about the black/red discrepancy; I don't think we should assume that the red is somehow the non-plus-ultra the black should aspire to match. TTFAutohint simply has different default settings than the Windows renderer. Let's compare TTFAutohint settings and judge them on their own merit.

I'm on Linux and as far as I can tell, any hints get ignored anyway. Not sure about OS X. On Windows, I get the following results for the different values of this key TTFAutohint setting:
```
 -a, --stem-width-mode=S    select stem width mode for grayscale, GDI 
                            ClearType, and DW ClearType, where S is a
                            string of three letters with possible values
                            `n' for natural, `q' for quantized, and `s'
                            for strong (default: qsq)
```


![image](https://user-images.githubusercontent.com/554383/94886471-1e63ec00-0428-11eb-84e4-a912d7c1deca.png)

The differences are subtle; the first one I don't like because letters like "u" are stunted; the second one is like the third but a bit darker, but both have a crippled "v". The last one (no hints) seems to have no obvious issues (yay!) but the letters look pretty anemic.

I think grayscale and GDI are only WinXP and earlier? Either way, I can't test them on Browserstack, so :man_shrugging: 
Then there are other TTFAutohint parameters I don't understand how to use properly. Speaking of which – I have no idea what settings to use to reproduce the FontSquirrel output; again :man_shrugging:  ... I'm sure someone with more experience would be able to achieve much better results!

I wonder if @davelab6 might have any font production advice for us? Or perhaps even an interest in commissioning a pro to clean this up for inclusion in GF? :)

**Comment by Discordius** [comment]
Just to follow up on this, do we want to merge any of these options?

**Comment by adamschwartz** [comment]
I would gladly accept a PR at this point. Perhaps you, @skosch, @davelab6 can work together on this?

**Comment by Discordius** [comment]
@skosch: Are you up for making a PR? I can probably figure out how to set up TTFAutohint and stuff, but seems like you basically had the files ready.

**Comment by skosch** [comment]
Yes and no. Do we have a consensus on what the best option is yet? I could certainly run TTFAutohint on the ttfs in the source folder to remove existing hints, which will solve this issue but not #21.

The right approach would probably be to modernize this whole repo, and to simply leave out the hinting in future builds, or to  always run TTFAutohint with the decided-upon settings:
- convert the sources to UFO format (@adamschwartz what software did you use to make these fonts?)
- follow a standard folder structure like [this one](https://github.com/unified-font-repository/Unified-Font-Repository)
- implement a reproducible build process (this could probably just be a [fontmake](https://github.com/googlefonts/fontmake) invocation)
- get rid of the lining/oldstyle number file split, and use the `onum` OpenType feature instead

I think this would be a worthwhile effort -- this is a very popular repo, and it would make it easier for others to contribute in the future (more language support, more italic weights, etc.).

Unfortunately I've never done any of these things myself, and I only have Linux and FontForge at my disposal, so I'm reluctant to spend a lot of time on this only to risk creating new problems. @Discordius depending on what kind of work you are willing to pay out the bounty for, perhaps there are others who could do this; I could certainly ask around.

**Comment by dpk** [comment]
I’m making progress on doing something like what @skosch suggests here: <https://github.com/dpk/et-book/tree/2.0>

**Comment by airza** [comment]
If you like me would like this issue fixed on your own website, @dpk 's fix works correctly. Steps:

- clone the [repo](https://github.com/dpk/et-book/tree/2.0)
- install [adobe's font tools](https://github.com/adobe-type-tools/afdko) using a venv in the repo
- activate the venv
- install [ttf2woff](https://github.com/adobe-type-tools/afdko)
- run `make`
- if make doesn't work, crawl in a hole and die (it worked for me after trial and error and I'm not sure what to do if it doesn't)
- otherwise, the new fonts should work. can see in action [here](https://www.airza.net/2025/05/20/dont-trade-life-for-sand) which appears to fix the issue!
 
Thanks so much to @dpk for the end of this madness!!!!

**Comment by Mike-Arnold** [comment]
The lowercase 'w' in the ttf file(s) appears to simply be lower than the 'v' by 5 units. A transformation of moving it up a bit fixed it for me.

The measurements found by checking with python fontTools:

v height: 882
w height: 882

v bottom: -23
w bottom: -28

v top: 859
w top: 854

Steps to fix:

- Open (say) [et-book-roman-line-figures.ttf](https://github.com/edwardtufte/et-book/blob/gh-pages/et-book/et-book-roman-line-figures/et-book-roman-line-figures.ttf) in FontForge.
- Open the 'w' glyph.
- Element -> Transformations -> Transform...
- select Move... (the default) and set X = 0, Y= 5 and hit OK
- File -> Generate Fonts... and make sure it saves as an otf.

**Comment by goodguypeci-design** [comment]
Hi @Discordius — is the $500 offer still current?

I have a Windows machine available and I'm reproducing the issue against the current fonts at several sizes in Chromium. I also found the older 2.0 work in PR #25 and don't want to duplicate it or step on that contribution.

If the offer is still open, would a focused, reproducible fix against `gh-pages` qualify if it includes the font source/build change and before/after Windows screenshots, or are you only interested in the broader 2.0 modernization now? PayPal or bank transfer would both work for me.

**Comment by goodguypeci-design** [comment]
@Discordius, I went ahead and prepared the full end-to-end fix you described in #31. If your offer still stands, it is ready for your Windows check; @adamschwartz, it is also ready for review.

It keeps the existing RomanLF/RomanOSF faces and numeral behavior, and changes only the lowercase `w` hint instructions in their TTF/WOFF files. On Windows Chromium and Firefox the oversized/dark `w` spike disappears at 12px, 17px, and 22px; widths stay the same, and a pixel comparison across `a-z`, `A-Z`, and `0-9` found no changes outside lowercase `w`.

The PR includes the exact results and native-resolution before/after overlays. Could you take a Windows pass when you get a chance?

**Comment by snesan821** [comment]
@Discordius Is the $500 reward still available for additional work, or already allocated to the earlier contributors? I see the existing fixes in #29, #30 and #31 and would first look for a remaining reproducible gap rather than submit another overlapping patch. My work would be AI-led for Sri, with the actual rendering checks documented. If there is still a paid gap, could you confirm the acceptance criteria and whether PayPal can be paid on your verification? This is an availability inquiry, not a bounty claim.

**Comment by airza** [comment]
It seems unlikely the bounty is still open. Anyone with a frontier llm plan can fix this themselves now.
